### ISSUE TITLE
Small-datasets cap: honour SMALLSHP so capped datasets stop re-simulating, honour PYAUTO_DISABLE_JAX in apply_sparse_operator, cap pixelization mesh shapes with the data

### ISSUE THREAD

**Issue by Jammy2211** [Jammy2211/PyAutoArray#528]
## Overview

Phase 8c (library leg) of the `ci-timing-fast-tests` epic (PyAutoMind `draft/feature/pyautoheart/ci_timing_fast_tests_epic.md`): the three PyAutoArray findings of the phase-8 diagnosis of the user-workspace smoke timings (autolens_workspace#536), each with a measured diff. Lands after the PyAutoNerves issue that adds `disable_jax()` and the `SMALLSHP` card.

1. **Capped interferometer / multi_dataset datasets re-simulate every run** (`draft/bug/autoarray/small_datasets_cap_stamp_stops_resimulation.md`). `_is_capped_at_the_current_cap` requires `SMALLDAT = T` **and** a measured `data.fits` shape equal to `SMALL_DATASETS_SHAPE_NATIVE`; interferometer data is `(n_visibilities, 2)` and multi_dataset prefixes its FITS, so neither can corroborate and both are deleted and re-simulated on every run — 5.2–6.5 s per script, ~26 s per autolens_workspace CI run, ~28 s per autogalaxy_workspace run, plus every local run.
2. **`apply_sparse_operator(use_jax=True)` ignores `PYAUTO_DISABLE_JAX=1`** (`draft/bug/autoarray/sparse_operator_ignores_disable_jax.md`): 2.3–3.2 s of JAX JIT per interferometer script under a harness that asked for NumPy; the workspace guides document the variable as the global switch.
3. **Mesh `shape=` does not honour `PYAUTO_SMALL_DATASETS`** the way `Grid2D.uniform` and `Mask2D.circular` do (`draft/feature/autoarray/mesh_shape_honours_small_datasets_cap.md`): a capped run reconstructs 1600–2500 source pixels from ~80 image pixels. Capping mesh shapes with the same `respect_small_datasets=True` escape hatch subsumes the four script guards HowToLens#77 / HowToGalaxy#73 added and reaches ~28 further call sites. Measured on those four scripts: 43.6 → 17.3 s.

## Plan

- `autoarray/util/dataset_util.py`: read the `SMALLSHP` card (`_small_datasets_shape_on_disk`); in `_is_capped_at_the_current_cap`, when the card is present compare it to today's cap and skip shape corroboration; absent card → today's behaviour. Add the `{waveband}_data.fits` / channel path resolution the file's own "Known gap" section names, so multi_dataset and datacube get a verdict. Safety property to test: the change only ever moves a dataset from *delete* to *keep*, and only when the writer recorded exactly today's cap.
- `apply_sparse_operator` (interferometer and imaging datasets): `if disable_jax(): use_jax = False` via the new `autonerves.test_mode.disable_jax()` predicate, with the rationale comment; unit test.
- Mesh cap: in the rectangular mesh / `Overlay` image-mesh constructors, cap `shape` to `SMALL_DATASETS_SHAPE_NATIVE` under `PYAUTO_SMALL_DATASETS=1` with `respect_small_datasets=True` (mirroring `Grid2D.uniform`); unit tests for both regimes.
- Validate: PyAutoArray unit tests; then, with the branch installed, the autolens_workspace_test and autogalaxy_workspace_test smoke gates (every relational assertion must still hold — the `_test` JAX scripts run uncapped so their pins are unaffected, but capped gate entries with explicit meshes will see smaller meshes) and the four workspace scripts before/after under the smoke profile.

<details>
<summary>Detailed implementation plan</summary>

### Branch
`claude/ci-test-timing-epic-ke2lul` (library-first: PyAutoNerves' PR merges before this one).

### Diffs (as measured by the phase-8 executor)

```diff
--- a/autoarray/util/dataset_util.py
+++ b/autoarray/util/dataset_util.py
 def _is_capped_at_the_current_cap(dataset_path):
-    return (
-        _small_datasets_stamp_on_disk(dataset_path) is True
-        and _is_small_datasets_on_disk(dataset_path)
-    )
+    if _small_datasets_stamp_on_disk(dataset_path) is not True:
+        return False
+    recorded = _small_datasets_shape_on_disk(dataset_path)
+    if recorded is not None:
+        return recorded == SMALL_DATASETS_SHAPE_NATIVE
+    return _is_small_datasets_on_disk(dataset_path)
```

```diff
--- a/autoarray/dataset/interferometer/dataset.py
+++ b/autoarray/dataset/interferometer/dataset.py
@@ def apply_sparse_operator(self, ..., use_jax: bool = False):
+        # `PYAUTO_DISABLE_JAX=1` is a harness-level override, not a preference: it is
+        # the documented way to force the NumPy path. An explicit `use_jax=True` in a
+        # script must not defeat it, or the smoke profile pays a JIT compile for a
+        # backend it asked to disable.
+        if disable_jax():
+            use_jax = False
```
(identically in `imaging/dataset.py`).

Mesh cap: the same pattern as `uniform_2d.py:570` (`if respect_small_datasets and small_datasets(): shape = min(shape, SMALL_DATASETS_SHAPE_NATIVE)` per axis), in the `Rectangular*` mesh constructors and `image_mesh.Overlay`.

### Acceptance
- Unit tests green (new tests for all three).
- With the branch installed: both `_test` gates green; the capped interferometer / multi_dataset scripts no longer re-simulate on a second run (`should_simulate` False); the four HowTo scripts unchanged in time with their guards, and the same without them.
</details>
